extension/commands/testSmartContractCommand.ts

```diff
@@ -125,36 +125,44 @@ export async function testSmartContract(allContracts: boolean, chaincode?: Insta
     let functionalTestsDirectory: string;
     let localSmartContractDirectory: string;
     let testFileSuffix: string;
-    if (contractLang === 'golang' || contractLang === 'java') {
+    let workspaceProjectName: string;
+    if (contractLang === 'golang' ) {
         outputAdapter.log(LogType.ERROR, `Automated functional tests for a smart contract project written in ${contractLang} are currently not supported.`);
         return;
     } else {
-        // Node smart contract: package.json must exist, chaincode and project names must match.
-        const packageJsonFile: string = path.join(chosenPackageFolder.uri.fsPath, 'package.json');
-        const packageJSONBuffer: Buffer = await fs.readFile(packageJsonFile);
-        const packageJSONObj: any = JSON.parse(packageJSONBuffer.toString('utf8'));
-        const replaceRegex: RegExp = /@.*?\//;
-        let workspaceProjectName: string = packageJSONObj.name;
-        workspaceProjectName = workspaceProjectName.replace(replaceRegex, '');
-        if (workspaceProjectName === chaincodeName) {
-            // Smart contract is open in workspace
-            functionalTestsDirectory = path.join(chosenPackageFolder.uri.fsPath, 'functionalTests');
-            localSmartContractDirectory = path.join(chosenPackageFolder.uri.fsPath, );
-            const languagesToSuffixes: object = {
-                JavaScript: 'js',
-                 TypeScript: 'ts'
-             };
-            const testLanguageItem: LanguageQuickPickItem = await UserInputUtil.showLanguagesQuickPick('Choose preferred test language', [], Object.keys(languagesToSuffixes));
-            if (!testLanguageItem) {
+        const languagesToSuffixes: object = {
+            JavaScript: 'js',
+            TypeScript: 'ts',
+            Java:       'java'
+        };
+        if (contractLang === 'java') {
+            testLang = 'Java';
+            functionalTestsDirectory = path.join(chosenPackageFolder.uri.fsPath, 'src', 'test', 'java', 'org', 'example');
+            workspaceProjectName = chaincodeName;
+        } else {
+            // Node smart contract: package.json must exist, chaincode and project names must match.
+            const packageJsonFile: string = path.join(chosenPackageFolder.uri.fsPath, 'package.json');
+            const packageJSONBuffer: Buffer = await fs.readFile(packageJsonFile);
+            const packageJSONObj: any = JSON.parse(packageJSONBuffer.toString('utf8'));
+            const replaceRegex: RegExp = /@.*?\//;
+            workspaceProjectName = packageJSONObj.name;
+            workspaceProjectName = workspaceProjectName.replace(replaceRegex, '');
+            if (workspaceProjectName === chaincodeName) {
+                // Smart contract is open in workspace
+                functionalTestsDirectory = path.join(chosenPackageFolder.uri.fsPath, 'functionalTests');
+                localSmartContractDirectory = path.join(chosenPackageFolder.uri.fsPath, );
```

Comment:
I've just noticed this has been in our code for a while - can we not just say `localSmartContractDirectory = chosenPackageFolder.uri.fsPath`?